src/neptune/metadata_containers/run.py

```diff
@@ -451,6 +461,43 @@ def _write_initial_attributes(self):
             # upload default sources ONLY if creating a new run
             upload_source_code(source_files=self._source_files, run=self)
 
+        if self._dependencies:
+            self._track_dependencies()
+
+    def _track_dependencies(self) -> None:
+        if self._dependencies == "infer":
+            from neptune.internal.utils.logger import logger
+
+            logger.info("INFO: Calling pipreqs to generate requirements based on project imports.")
+            # using pipreqs here
+            try:
+                proc = subprocess.Popen(["pipreqs", "--print", "."], stdout=subprocess.PIPE)
+                proc.wait()
+                dependencies_str = proc.communicate()[0].decode(encoding="utf-8")
+            except subprocess.SubprocessError as exp:
+                logger.error("ERROR: {}".format(exp))
+                dependencies_str = ""
+
+            if not dependencies_str:
+
+                logger.error("ERROR: Could not generate requirements. Call to 'pipreqs' returned an empty string.")
+                return
+
+            self["source_code/requirements.txt"].upload(File.from_content(dependencies_str))
```

Comment:
`from_content` has a key parameter `extension` that could be set to `txt`. Ensure that this will not produce `requirements.txt.txt` file after downloading. Probably we should rename the namespace to just `requirements`.

Reply:
[from docs of `from_content`]
extension – Extension of the created file. File will be used for interpreting the type of content for visualization. If None it will be bin for binary content and txt for text content. Defaults to None.

Since we upload a string, then I guess there's no need to add it explicitly and just rename namespace to requirements

Reply:
Ok